File: src/main/java/io/github/vehkiya/data/model/domain/Item.java
package io.github.vehkiya.data.model.domain;

import lombok.Data;
import lombok.experimental.Accessors;

import java.util.Set;

@Data
@Accessors(chain = true, fluent = true)
public class Item {

    private String name;

    private Recipe recipe;

    private Set<Recipe> alternativeRecipes;

}

File: src/main/java/io/github/vehkiya/data/model/persistence/Item.kt
package io.github.vehkiya.data.model.persistence

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "Item")
data class Item(
    var name: String, @Id @GeneratedValue var id: Long? = null
)